Mixed sign comparison

As of #69 (in particular [this change I requested](https://github.com/google/libsxg/pull/69/files/9a5e2025b2e180c599678c762c57dbde809a2916#r554246842)), we get this warning when compiling against OpenSSL (not BoringSSL):

```
/usr/local/google/home/twifkak/devel/libsxg/src/sxg_codec.c: In function ‘sxg_base64encode_bytes’:
/usr/local/google/home/twifkak/devel/libsxg/src/sxg_codec.c:52:18: warning: comparison of integer expressions of different signedness: ‘int’ and ‘size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
   52 |   if (out_length < length) return false;
      |                  ^
```

Fix this.

--- a/src/sxg_codec.c
+++ b/src/sxg_codec.c
@@ -49,7 +49,7 @@ bool sxg_base64encode_bytes(const uint8_t* src, size_t length,
   const size_t offset = dst->size;
   // 3-byte blocks to 4-byte, rounded up
   const EVP_ENCODE_BLOCK_T out_length = 4 * ((length + 2) / 3);
-  if (out_length < length) return false;
+  if (out_length < 0 || (size_t)out_length < length) return false;
 
   return sxg_buffer_resize(offset + out_length, dst) &&
          EVP_EncodeBlock(dst->data + offset, src, length) == out_length;

